Questionarium/src/main/java/repository/QuestionRepositoryImp.java

```diff
@@ -0,0 +1,105 @@
+package repository;
+
+import model.Question;
+import repository.dao.QuestionRepository;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QuestionRepositoryImp implements QuestionRepository {
+    private Connection connection;
+
+    public QuestionRepositoryImp(Connection connection) {
+        this.connection = connection;
+    }
+
+    private final String findById = "SELECT * FROM Questions where id=?";
+    private final String saveQuestion = "INSERT INTO Questions VALUE (?,?,?)";
+    private final String findByTopic = "SELECT * FROM Questions where topic=?";
+    private final String updateText = "UPDATE Questions SET text=? where id=?";
+    private final String deleteById = "DELETE FROM Questions where id=?";
+
+
+    @Override
+    public Question get(int id) {
+        try {
+            PreparedStatement preparedStatement = this.connection.prepareStatement(findById);
+            preparedStatement.setInt(1, id);
+            ResultSet question = preparedStatement.executeQuery();
+            question.next();
+            return Question.builder()
+                    .id(question.getInt("id"))
+                    .text(question.getString("text"))
+                    .topic(question.getString("topic"))
+                    .build();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void save(Question question) {
+        try {
+            PreparedStatement preparedStatement = this.connection.prepareStatement(saveQuestion);
+            preparedStatement.setInt(1, question.getId());
+            preparedStatement.setString(2, question.getText());
+            preparedStatement.setString(3, question.getTopic());
+            if (preparedStatement.executeUpdate() != 1) {
+                throw new SQLException("Something went wrong");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void update(Question question) {
+        try {
+            PreparedStatement preparedStatement = this.connection.prepareStatement(updateText);
+            preparedStatement.setString(1, question.getText());
+            preparedStatement.setInt(2, question.getId());
+            if (preparedStatement.executeUpdate() != 1) {
+                throw new SQLException("Something went wrong");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void delete(int id) {
+        try {
+            PreparedStatement preparedStatement = this.connection.prepareStatement(deleteById);
+            preparedStatement.setInt(1, id);
+            if (preparedStatement.executeUpdate() != 1) {
```

Comment:
if the query is not executed properly it is not an exception, it is developer's mistake and you do no need to throw an exception because of it